src/python/pants/build_graph/target.py

```diff
@@ -755,30 +726,25 @@ def dependencies(self):
     return [self._build_graph.get_target(dep_address)
             for dep_address in self._build_graph.dependencies_of(self.address)]
 
-  def exports(self, dep_context):
-    """
-    :param dep_context: A DependencyContext with configuration for the request.
-
-    :return: targets that this target directly exports. Note that this list is not transitive,
-      but that exports are transitively expanded during the computation of strict_dependencies.
-    :rtype: list of Target
-    """
+  def export_addresses(self):
     exports = self._cached_exports_map.get(dep_context, None)
```

Comment:
`dep_context` is missing here.

It exists in order to "dependency inject" information about various target types that should affect the search in different ways... but that's a bit unfortunate. Having `Target` be aware of its various subclasses is also unfortunate. We used to have Target "labels" to guide this type of search.

I think that moving forward, because there will not just be one "BuildGraph", the dependency search will naturally be very, very type centric (cf RuleGraph)... which I hope will make things like this easier... not harder.

Reply:
Yeah. I lost that in an extraction. Thanks for the catch. I also think that things like this would be less awkward in the new model. For one, we won't need to have these local caches for things.